syntax = "proto3";

option java_multiple_files = true;
option java_package = "sr.grpc.gen";
option java_outer_classname = "SubscriptionProto";

package subscription;

service Subscription {
  rpc InformAboutMeeting(SubscriptionRequest) returns (stream Event);
  rpc Ping(PingMsg) returns (PingMsg);
}

message SubscriptionRequest {
  Category category = 1;
}

enum Category {
	BUSINESS = 0;
	FASHION = 1;
	IT = 2;
	MUSIC = 3;
	SPORT = 4;	
 }

message Event {
  string name = 1;
  Category category = 2;
  string date = 3;
  repeated string cities = 4;  
}

message PingMsg {
}